gptlatextranslator/GPTTranslator.py

```diff
@@ -3,21 +3,65 @@
 import warnings
 from typing import List
 
+# For more info on available models, see https://platform.openai.com/docs/models/overview
+MODELS = [
+    "gpt-3.5-turbo",
+    "gpt-3.5-turbo-16k",
+    "gpt-4",
+    "gpt-4-32k"
+  ]
+
+def get_default_max_input_tokens(model_name: str) -> int:
+    """Returns the default maximum number of tokens that can be sent to the model, before running intro troubles."""
+    max_tokens = None
+
+    if model_name == "gpt-3.5-turbo":
+        max_tokens = 4096
+    elif model_name == "gpt-3.5-turbo-16k":
+        max_tokens = 16384
+    elif model_name == "gpt-4":
+        max_tokens = 8192
+    elif model_name == "gpt-4-32k":
+        max_tokens = 32768
+    else:
+        raise ValueError(f"Unknown model name '{model_name}'.")
+
+    # 0.31 is a magic number that allows us enough space to add the system context and the translation prompt, while also taking into account the output of the model.
+    max_input_tokens = int(max_tokens * 0.31)
```

Comment:
I guess this comes from the old `1300` `max_input_tokes` for the `4096` of the `gpt-3.5-turbo` model, right?

Reply:
Yes, I thought that the number `1300` was working very well for the `4096` limitation in the  `gpt-3.5-turbo` model, so I extrapolated similar numbers for the other models. For example, the `gpt-3.5-turbo-16k` has a max tokens limitation of `16384`, so the max input tokens would be `5079`.